dGame/dComponents/PropertyEntranceComponent.cpp

```diff
@@ -74,261 +67,222 @@ void PropertyEntranceComponent::OnEnterProperty(Entity* entity, uint32_t index,
+	auto propertyLookup = Database::CreatePreppedStmt(query);
 
-		friendsListQuery->setUInt(1, character->GetID());
-		friendsListQuery->setUInt(2, character->GetID());
+	const auto searchString = "%" + searchText + "%";
+	propertyLookup->setUInt(1, this->m_MapID);
+	propertyLookup->setString(2, searchString.c_str());
+	propertyLookup->setString(3, searchString.c_str());
+	propertyLookup->setString(4, searchString.c_str());
+	propertyLookup->setInt(5, sortMethod == SORT_TYPE_FEATURED || sortMethod == SORT_TYPE_FRIENDS ? (uint32_t)PropertyPrivacyOption::Friends : (uint32_t)PropertyPrivacyOption::Public);
+
+	auto propertyIdRes = propertyLookup->executeQuery();
 
-		auto friendsListQueryResult = friendsListQuery->executeQuery();
+	std::vector<uint32_t> propertyIds{};
 
-		while (friendsListQueryResult->next()) {
-			auto playerIDToConvert = friendsListQueryResult->getInt(1);
-			friendsList = friendsList + std::to_string(playerIDToConvert) + ",";
+	while (propertyIdRes->next()) {
+		if (sortMethod == SORT_TYPE_FRIENDS || sortMethod == SORT_TYPE_FEATURED) {
+			if (m_UserFriendMap[(uint32_t)requestor->GetObjectID()].find(propertyIdRes->getUInt(2)) == m_UserFriendMap[(uint32_t)requestor->GetObjectID()].end()) {
+				continue;
+			}
 		}
-		// Replace trailing comma with the closing parenthesis.
-		if (friendsList.at(friendsList.size() - 1) == ',') friendsList.erase(friendsList.size() - 1, 1);
-		friendsList += ") ";
 
-		// If we have no friends then use a -1 for the query.
-		if (friendsList.find("()") != std::string::npos) friendsList = " AND p.owner_id IN (-1) ";
+		propertyIds.push_back(propertyIdRes->getUInt(1));
+	}
 
-		orderBy += friendsList + "ORDER BY ci.name ASC ";
+	delete propertyIdRes;
+	propertyIdRes = nullptr;
 
-		delete friendsListQueryResult;
-		friendsListQueryResult = nullptr;
+	delete propertyLookup;
+	propertyLookup = nullptr;
 
-		delete friendsListQuery;
-		friendsListQuery = nullptr;
-	} else if (sortMethod == SORT_TYPE_RECENT) {
-		orderBy = "ORDER BY p.last_updated DESC ";
-	} else if (sortMethod == SORT_TYPE_REPUTATION) {
-		orderBy = "ORDER BY p.reputation DESC, p.last_updated DESC ";
-	} else {
-		orderBy = "ORDER BY p.last_updated DESC ";
-	}
-	return base + orderBy + (wantLimits ? "LIMIT ? OFFSET ?;" : ";");
+	return propertyIds;
 }
 
+
 void PropertyEntranceComponent::OnPropertyEntranceSync(Entity* entity, bool includeNullAddress, bool includeNullDescription, bool playerOwn, bool updateUi, int32_t numResults, int32_t lReputationTime, int32_t sortMethod, int32_t startIndex, std::string filterText, const SystemAddress& sysAddr) {
+	std::vector<PropertyData> entries{};
 
-	std::vector<PropertySelectQueryProperty> entries{};
-	PropertySelectQueryProperty playerEntry{};
+	if (m_UserFriendMap.find((uint32_t)entity->GetObjectID()) == m_UserFriendMap.end()) {
```

Comment:
I don't believe you can cache the friends list here unfortuntately, because a player can make more friends in this world and then re-open the menu to try and visit their new friends property.  Is this scenario taken into account somewhere?